presto-client/src/main/java/io/prestosql/client/Row.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.client;
+
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public final class Row
+{
+    private final List<RowField> fields;
+
+    private Row(List<RowField> fields)
+    {
+        this.fields = ImmutableList.copyOf(requireNonNull(fields, "fields is null"));
```

Comment:
rnn not needed (done by copyOf)

Reply:
i have a nicer message, as i say what param is null (here there is only one, but you i do not want to change this line, if i add annother one)

Reply:
True. I put a comment here mostly because we typically not do `rnn` if there is `copyOf` already.

Reply:
I think this depends on "we" :)